src/memex/search/index.py

```diff
@@ -92,25 +119,16 @@ class BatchIndexer(object):
     the search index.
     """
 
-    def __init__(self, session, es_client, request):
+    def __init__(self, session, es_client, request, target_index=None):
         self.session = session
         self.es_client = es_client
         self.request = request
 
         # By default, index into the open index
-        self._target_index = self.es_client.index
-
-    def index_all(self):
```

Comment:
I see this retrying functionality more part of a batch indexer, mainly because of the "batch" part of it where I don't want to care about transient errors of individual items in the batch